overlord/snapshotstate/backend/backend.go

```diff
@@ -396,6 +396,7 @@ func addDirToZip(ctx context.Context, snapshot *client.Snapshot, w *zip.Writer,
 		"--create",
 		"--sparse", "--gzip",
 		"--format", "gnu",
+		"--warning", "no-file-ignored",
```

Comment:
I wonder if we should silently ignore the files instead of logging which were ignored.

Also, from my silly attempts with sockets in a directory, tar the same command line (but without no-file-ignored), does not exit with non-0. Then I proceeded to create sockets and fifos in $SNAP_DATA and $SNAP_COMMON of hello-world, and snap save worked. I think there may have been more errors logged by tar, but they were not included in the error output of the task.

Reply:
You are right, the sockets ignored was hiding the actual cause of the error

```
tar: ./171/.datalayer/frontend/2069: socket ignored
tar: ./171/.datalayer/backend/2070: socket ignored
tar: ./171/package-run/rexroth-automationcore/package-www-fileserver.sock: socket ignored
tar: ./171: file changed as we read it
1
```

We found the cause on our side for this. But it would be helpful to have the "real" error cause in the log of the snap change.